internal/parser/argument_parsers.go

```diff
@@ -68,23 +68,34 @@ func parseSaddFunction(command string, arguments []string, cacheMap map[string]*
 		return keyValueObject
 	}
 
-	for _, newValue := range arguments[1:] {
-		for _, value := range keyValueObject.value {
-			if value == newValue { // value already in the set
-				continue
-			}
+	set := make(map[string]struct{})
```

Comment:
I love this PR. I was even thinking of implementing the "PING" command and you've done it already. However, why did you prefer to use a map of struct values for the set duplicate values search? Why not use a boolean value (if the set value already exists then its value in the map should be true)? Is there any reason why you chose this path? Once again, I like that you're now being performance-aware in your implementations. You are doing well.

Reply:
Thanks a lot, boss. Learning all these from you. There's a reason I chose map of struct values instead of boolean. I implemented it using booleans at first and it worked well so i was thinking of improving the performance.
The empty structs do not assign memory and they occupy zero bytes of storage. 
Also came across an article that explained this [here](https://itnext.io/set-in-go-map-bool-and-map-struct-performance-comparison-5315b4b107b) `map[]struct{}` is **5%** faster in time and **10%** less memory consumption comparing to `map[]bool` when it comes to a big Set.